daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -38,6 +39,56 @@ sealed abstract class SBuiltin(val arity: Int) {
   /** Execute the builtin with 'arity' number of arguments in 'args'.
     * Updates the machine state accordingly. */
   def execute(args: util.ArrayList[SValue], machine: Machine): Unit
+
+  private def typeMismatch(i: Int, expected: String, found: SValue) =
+    crash(s"type mismatch for arg($i) in $productPrefix: expected $expected, found: $found.")
+
+  @inline
```

Comment:
I'm not sure the `@inline` annotation can do much for a _merely_ `protected` method on an `abstract` class. Maybe if they could be made `final`...